### PR TITLE
Admin Notes: Add CRUD and Data Stores

### PR DESCRIPTION
**Pull request by allendav**
See #176 

Ready for testing.

This PR implements the data stores for the admin notes. It does not include the REST API interfaces (that will be in a subsequent PR) as this PR is already too big. Unit tests will also be provided in a subsequent PR once the unit test PR has landed - in the mean time, to test, you'll also need this plugin: https://github.com/Automattic/wc-admin-notes-test-bench

To test:
- Install and activate wc-admin using this branch
- Install and activate the test bench plugin
- Go to wp-admin > Tools > WC Admin Notes Test Bench
- Add a new note, e.g.

screen shot 2018-09-26 at 10 44 54 am

- After adding, verify it appears in the test bench, e.g.:

screen shot 2018-09-26 at 10 46 03 am

- You may also verify the note and action in SQL, e.g.:

screen shot 2018-09-26 at 10 47 07 am

Enjoy!

### CODE REVIEW

File: includes/class-wc-admin-note.php [hunk 1]
```diff
@@ -0,0 +1,463 @@
+<?php
+/**
+ * WooCommerce Admin (Dashboard) Notes.
+ *
+ * The WooCommerce admin notes class gets admin notes data from storage and checks validity.
+ *
+ */
+defined( 'ABSPATH' ) || exit;
+
+class WC_Admin_Note extends WC_Data {
```
Comment (claudiosanches) [inline includes/class-wc-admin-note.php]: Missing docblock.

File: includes/class-wc-admin-note.php [hunk 2]
```diff
@@ -0,0 +1,463 @@
+<?php
+/**
+ * WooCommerce Admin (Dashboard) Notes.
+ *
+ * The WooCommerce admin notes class gets admin notes data from storage and checks validity.
+ *
```
Comment (claudiosanches) [inline includes/class-wc-admin-note.php]: Missing @package.

File: includes/class-wc-admin-note.php [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+ */
+defined( 'ABSPATH' ) || exit;
+
+class WC_Admin_Note extends WC_Data {
+
+	// Note types
+	const E_WC_ADMIN_NOTE_ERROR = 'error';
```
Comment (claudiosanches) [inline includes/class-wc-admin-note.php]: Missing comments for all conts here, also all inline comments need to stop with `.`.

File: includes/class-wc-admin-note.php [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+	|
+	*/
+
+	/**
+	 * Set note name.
+	 *
+	 * @param string $name Note name.
+	 */
+	public function set_name( $name ) {
+		// Don't allow empty names
```
Comment (claudiosanches) [inline includes/class-wc-admin-note.php]: Missing period.

**Review by claudiosanches — CHANGES_REQUESTED** [review]
There is a lot of coding standards issues to fix. I pointed a few, but probably better try PHPCS.

**Comment by allendav** [comment]
@claudiosanches wrote:

> There is a lot of coding standards issues to fix. I pointed a few, but probably better try PHPCS.

Oops! Will fix asap!

**Comment by allendav** [comment]
@claudiosanches All the coding standards issues are fixed now.

File: includes/class-wc-admin-note.php (hunk 1 above)
Comment (allendav) [reply includes/class-wc-admin-note.php]: Fixed in https://github.com/woocommerce/wc-admin/pull/464/commits/4433670cb7eb3ecf2b4d05ed34b83b6ab4e6dea8

File: includes/class-wc-admin-note.php (hunk 2 above)
Comment (allendav) [reply includes/class-wc-admin-note.php]: Fixed in https://github.com/woocommerce/wc-admin/pull/464/commits/4433670cb7eb3ecf2b4d05ed34b83b6ab4e6dea8

File: includes/class-wc-admin-note.php (hunk 4 above)
Comment (allendav) [reply includes/class-wc-admin-note.php]: Fixed in https://github.com/woocommerce/wc-admin/pull/464/commits/4433670cb7eb3ecf2b4d05ed34b83b6ab4e6dea8

File: includes/class-wc-admin-note.php (hunk 3 above)
Comment (allendav) [reply includes/class-wc-admin-note.php]: Fixed in https://github.com/woocommerce/wc-admin/pull/464/commits/0aaca72b0d2d4adcf7955a8264337ade36232de0

**Comment by allendav** [comment]
@timmyc - FWIW, my PHPCS setup broke and stopped running in Atom. I fixed it and everything @claudiosanches mentioned (and a few more) lit right up like a Christmas tree :)

**Comment by claudiosanches** [comment]
@timmyc @allendav Help with Coding Standards is coming: https://github.com/woocommerce/wc-admin/pull/488

**Review by claudiosanches — APPROVED** [review]
Works great, good work.

**Pull request merged**